camel/memories/base.py

```diff
@@ -70,6 +47,93 @@ def write_record(self, record: MemoryRecord) -> None:
 
     @abstractmethod
     def clear(self) -> None:
-        r"""Clears all messages from the memory.
+        r"""Clears all messages from the memory."""
+        pass
+
+
+class BaseContextCreator(ABC):
+    r"""An abstract base class defining the interface for context creation
+    strategies.
+
+    This class provides a foundational structure for different strategies to
+    generate conversational context from a list of context records. The
+    primary goal is to create a context that is aligned with a specified token
+    count limit, allowing subclasses to define their specific approach.
+
+    Subclasses should implement the `token_counter`, `token_limit`, and
```

Comment:
nit: `:obj:`token_counter``
```suggestion
    Subclasses should implement the `token_counter`, `token_limit`, and
```